src/mqt/bench/utils.py

```diff
@@ -313,38 +312,40 @@ def calc_qubit_index(qargs: list[Qubit], qregs: list[QuantumRegister], index: in
 def calc_supermarq_features(
     qc: QuantumCircuit,
 ) -> SupermarqFeatures:
-    qc = qc.remove_final_measurements(inplace=False)
-    qc = RemoveBarriers()(qc)
     connectivity_collection: list[list[int]] = []
     liveness_A_matrix = 0
     for _ in range(qc.num_qubits):
         connectivity_collection.append([])
 
-    for _, qargs, _ in qc.data:
-        liveness_A_matrix += len(qargs)
-        first_qubit = calc_qubit_index(qargs, qc.qregs, 0)
-        all_indices = [first_qubit]
-        if len(qargs) == 2:
-            second_qubit = calc_qubit_index(qargs, qc.qregs, 1)
-            all_indices.append(second_qubit)
-        for qubit_index in all_indices:
-            to_be_added_entries = all_indices.copy()
-            to_be_added_entries.remove(int(qubit_index))
-            connectivity_collection[int(qubit_index)].extend(to_be_added_entries)
+    for instruction, qargs, _ in qc.data:
+        if instruction.name != "barrier" and instruction.name != "measure":
+            liveness_A_matrix += len(qargs)
+            first_qubit = calc_qubit_index(qargs, qc.qregs, 0)
+            all_indices = [first_qubit]
+            if len(qargs) == 2:
+                second_qubit = calc_qubit_index(qargs, qc.qregs, 1)
+                all_indices.append(second_qubit)
+            for qubit_index in all_indices:
+                to_be_added_entries = all_indices.copy()
+                to_be_added_entries.remove(int(qubit_index))
+                connectivity_collection[int(qubit_index)].extend(to_be_added_entries)
 
     connectivity: list[int] = []
     for i in range(qc.num_qubits):
         connectivity.append(len(set(connectivity_collection[i])))
 
-    num_gates = sum(qc.count_ops().values())
+    count_ops = qc.count_ops()
+    num_gates = sum(count_ops.values()) - count_ops.get("measure") - count_ops.get("barrier")
     num_multiple_qubit_gates = qc.num_nonlocal_gates()
-    depth = qc.depth()
+    depth = qc.depth(lambda x: x[0].name != "barrier" and x[0].name != "measure")
     program_communication = np.sum(connectivity) / (qc.num_qubits * (qc.num_qubits - 1))
 
     if num_multiple_qubit_gates == 0:
         critical_depth = 0.0
     else:
-        critical_depth = qc.depth(filter_function=lambda x: len(x[1]) > 1) / num_multiple_qubit_gates
+        critical_depth = (
+            qc.depth(filter_function=lambda x: len(x[1]) > 1 and x[0].name != "barrier") / num_multiple_qubit_gates
```

Comment:
Why does this not exclude measurements?

Reply:
Because all local gates operating only on a single qubit are already filtered by the first condition `lambda x: len(x[1]) > 1`. Do you see a benefit in explicitly list the measurements as well for clarity?

Reply:
Ah. Missed that. Then consider this resolved 😎